src/base/bittorrent/portforwarderimpl.h

```diff
@@ -50,14 +51,17 @@ class PortForwarderImpl final : public Net::PortForwarder
     bool isEnabled() const override;
     void setEnabled(bool enabled) override;
 
-    void addPort(quint16 port) override;
-    void deletePort(quint16 port) override;
+    void setPorts(const QString &profile, QSet<quint16> ports) override;
+    void removePorts(const QString &profile) override;
 
 private:
     void start();
     void stop();
 
     CachedSettingValue<bool> m_storeActive;
     lt::session *const m_provider = nullptr;
-    QHash<quint16, std::vector<lt::port_mapping_t>> m_mappedPorts;
+
+    using PortMapping = QHash<quint16, std::vector<lt::port_mapping_t>>;  // <port, handles>
```

Comment:
The last comment on code of current implementation. I would use the following variables instead:
```c++
QHash<QString, quint16> m_portProfiles;
QHash<quint16, std::vector<lt::port_mapping_t>> m_forwardedPorts;
```

Reply:
I don't see the point. Why do you want to split information into 2 variables? In your suggestion, the "value" part of `m_portProfiles` is really the same as the "key" part of `m_forwardedPorts`. You'll probably need more lookups to get the mapping handles... it really feels unnatural to me.

In my implementation, the `QSet<quint16> m_forwardedPorts;` is unimportant and is just for bookkeeping whether the port is already taken and important handles/data should not be stored in it. This variable can be omitted by checking all ports in `m_portProfiles` but that is slower.

Reply:
> it really feels unnatural to me.

This is natural if you look at them in such a way that one represents the configuration and the other represents the state.
>m_forwardedPorts; is unimportant and is just for bookkeeping whether the port is already taken and important handles/data should not be stored in it. 

Ok.
>This variable can be omitted by checking all ports in m_portProfiles but that is slower.

I don't think it has any sensitive value, given that there are only a couple of ports here, besides, the actions affecting it are clearly not intended for any frequent use.